README.md

```diff
@@ -1,2 +1,37 @@
 # elasticsearch-templates
-templates, index templates, mappings, kibana configs for elasticsearch
+
+The repository contains scripts and sources to generate Elasticsearch templates
+that comply with Common Data Model.
+
+# Problem Statement
+
+We are trying to solve the problem of conflicts and inconsistencies in log data
+as collected by, and from, different subsystems stored together as a unified
+data set under one warehouse.
+
+# Namespace hierarchy
+
+Namespace hierarchy on the log metadata is the key concept.
+We use the Elasticsearch index templates and document mapping to cast the common
+metadata keys into usable documents.
+
+Namespace corresponds to a top-level JSON key of Elasticsearch document.
+Namespace is usually defined per individual app or subsystem, so that different
+applications/subsystems not conflict in various metadata fields.
```

Comment:
Add an example here of what the namespace filename and definition looks like (small snippet), and what the corresponding ES index template JSON looks like.